core/test/lib/libledger-test/CMakeLists.txt

```diff
@@ -5,22 +5,44 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 add_library(mangoose STATIC mongoose.c mongoose.h)
 
 add_library(ledger-test STATIC
-            EventLooper.cpp EventLooper.hpp EventThread.cpp EventThread.hpp
-            NativeThreadDispatcher.cpp NativeThreadDispatcher.hpp
-            NativePathResolver.cpp NativePathResolver.hpp
             CoutLogPrinter.cpp CoutLogPrinter.hpp
-            MongooseHttpClient.cpp MongooseHttpClient.hpp MongooseSimpleRestServer.cpp MongooseSimpleRestServer.hpp
-            route.cc route.h OpenSSLRandomNumberGenerator.cpp OpenSSLRandomNumberGenerator.hpp callbacks.cpp callbacks.hpp FakeWebSocketClient.cpp FakeWebSocketClient.h
-            FakeHttpClient.hpp FakeHttpClient.cpp
-            FakeUrlConnection.hpp FakeUrlConnection.cpp
+            CppHttpLibClient.hpp CppHttpLibClient.cpp
+            EventLooper.cpp EventLooper.hpp
+            EventThread.cpp EventThread.hpp
             ExplorerStorage.hpp ExplorerStorage.cpp
+            FakeHttpClient.hpp FakeHttpClient.cpp
+            FakeWebSocketClient.cpp FakeWebSocketClient.h
+            FilesystemUtils.cpp FilesystemUtils.hpp
             HttpClientOnFakeExplorer.hpp HttpClientOnFakeExplorer.cpp
-            UvThreadDispatcher.hpp UvThreadDispatcher.cpp
             MemPreferencesBackend.hpp MemPreferencesBackend.cpp
-            CppHttpLibClient.hpp CppHttpLibClient.cpp
-            FilesystemUtils.cpp FilesystemUtils.hpp
+            MongooseHttpClient.cpp MongooseHttpClient.hpp
+            MongooseSimpleRestServer.cpp MongooseSimpleRestServer.hpp
+            NativePathResolver.cpp NativePathResolver.hpp
+            NativeThreadDispatcher.cpp NativeThreadDispatcher.hpp
+            OpenSSLRandomNumberGenerator.cpp OpenSSLRandomNumberGenerator.hpp
+            UvThreadDispatcher.hpp UvThreadDispatcher.cpp
+            callbacks.cpp callbacks.hpp
+            route.cc route.h)
+
+add_library(ledger-test-proxy-http-client STATIC
```

Comment:
I would move all `ledger-test-proxy-http-client`-related files in a dedicated subdirectory with its own `CMakeLists.txt` cmake conf file. This way It will be more clear what is the scope of either `ledger-test-proxy-http-client`  and `ledger-test`

Reply:
done